util/stmtsummary/statement_summary.go

```diff
@@ -243,15 +245,17 @@ func (ssMap *stmtSummaryByDigestMap) AddStatement(sei *StmtExecInfo) {
 		prevDigest: sei.PrevSQLDigest,
 		planDigest: sei.PlanDigest,
 	}
+	// Calculate hash value in advance.
+	key.Hash()
```

Comment:
Why need this? Look like when `summaryMap.Put` and `summaryMap.Get` will call `key.Hash()` automatically.

Reply:
I want to move some work out of the lock. Otherwise, the time consumed by waiting for the lock will be long, and it lowers the concurrency.

Reply:
I added more comments.